# Introduction to short read NGS: Library construction, UCE capture and ddRADseq
Herpetology Group at the Natural History Museum, London | 
**Instructor: Jeff Streicher**

## Contents
**Overview** 
>This 4.5-week course aims to provide a conceptual overview of the basic laboratory methods used to generate short read Next Generation Sequencing (NGS) data on Illumina® platforms as well as the bioinformatics pipelines used to process resulting data for three types of analysis: (1) shotgun sequencing, (2) targeted sequence capture of ultraconserved elements (UCEs) and (3) double digest restriction-site associated DNA sequences (ddRADseq). At the end of the course, participants will have received an introduction to terminology, protocols and methods that can be further developed into a variety of natural history-associated research projects including phylogenomic and population genomic analyses.

**Unit 1: Introduction to short read sequencing and library preparation** 
>Click [here](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_01) to access Unit 1 content.

**Unit 2: Illumina libraries: de novo assembly and reference mapping** 
>Click [here](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_02) to access Unit 2 content. 

**Unit 3: Targeted sequence capture of ultraconserved elements (UCEs)** 
>Click [here](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_03) to access Unit 3 content.

**Unit 4: Double digest restriction-site associated DNA sequencing (ddRADseq)** 
>Click [here](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_04) to access Unit 4 content.

**Prerequisites**
>Basic familiarity with the molecular laboratory (i.e. pipetting, DNA extraction, PCR) and command line utilities (i.e. Linux and MacOS ‘bash’ shells or Windows PuTTy command line). Participants are encouraged to have an NHM user account on [Franklin](http://pgfsites.s3-website-eu-west-1.amazonaws.com/H21/franklin.html) or [Crop Diversity Cluster](https://www.cropdiversity.ac.uk/) for the bioinformatics labs. 
