# Bioinformatics Laboratory 2

>You will need a Unix Command Terminal or PuTTY interface to complete this lab.

## *de novo* assembly of libraries using phyluce

<details>
 <summary>Click to expand content!</summary>

 >There are several different ways to assemble contigs from your cleaned Illumina FASTQ data. In this module we will compare the results of two different assemblers that we will execute in [phyluce](https://phyluce.readthedocs.io/en/latest/index.html).

1. We will use the three sets of cleaned fastq.gz files from [Unit 1](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_01/Bioinformatics_Lab) that were downloaded from the NCBI [SRA](https://www.ncbi.nlm.nih.gov/sra) and originally sequenced for Streicher & Wiens ([2016](https://www.sciencedirect.com/science/article/abs/pii/S1055790316300495?via%3Dihub)) using a MiSeq and paired-end 300bp kit.  
```  
Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz
Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz
Cylindrophis_ruffus_FMNH_258674-READ-singleton.fastq.gz
``` 
>These files should be inside a directory called ```Unit_1/Data/clean-fastq/Cylindrophis_ruffus_FMNH_258674/split-adapter-quality-trimmed```
 
2. Activate phyluce
 ```  
  conda activate phyluce-1.7.1
 ``` 
3. To use phyluce to assemble reads into contigs, a configuration file is needed. The configuration file looks like this:

```
[samples]
Cylindrophis_ruffus_FMNH_258674:clean-fastq/Cylindrophis_ruffus_FMNH_258674/split-adapter-quality-trimmed/
  
```
The configuration file tells phyluce where to find the files that are to be *de novo* assembled. Our example only contains one sample, but you can have configuration files with as many samples/taxa/individuals as you like.

4. To make the configuration text file let's use the command line: 
 
 ```  
  cat > assembly.conf
 ```   
 Now paste the configuration text (from Step 3) into your terminal and then press CTRL + SHIFT + D. 
 
5. We are now ready to assemble the reads into contigs using velvet: 
 ```   
phyluce_assembly_assemblo_velvet \
    --conf assembly.conf \
    --output velvet-assemblies \
    --cores 12 
 ```
 >This will place the ouput files in a new directory called velvet-assemblies. On Franklin it should take about 3 mins to run. 
 
 6. Next, let's use the same configuration file to assemble the reads into contigs using spades: 
 ```   
phyluce_assembly_assemblo_spades \
    --conf assembly.conf \
    --output spades-assemblies \
    --cores 12 
 ``` 
  >This will place the ouput files in a new directory called spades-assemblies. On Franklin this may take ~10 mins to run.  

Once both assemblies have finished our ```Data``` directory should contain the following items: 
 
```
 assembly.conf         illumiprocessor.log                   raw-fastq
clean-fastq           phyluce_assembly_assemblo_spades.log  spades-assemblies
illumiprocessor.conf  phyluce_assembly_assemblo_velvet.log  velvet-assemblies
```

7. Once completed, both ```spades-assemblies``` and ```velvet-assemblies``` should contain two directories ```contigs``` and  ```Cylindrophis_ruffus_FMNH_258674```
 
The ```contigs``` directory contains a single FASTA file which is the output of the assembly. It will contain hundreds to thousands of sequences inferred by the assembly. 

Each line of the FASTA file will look something like this: 
 
 ```
>NODE_6_length_2008_cov_11.945212
TGGAGGCATAAAAGTGGCTGGGGGAAATGCGCTTTGTGGTGGAAGTGTGGTATATAAAGG
TTTGGCACTAAAAGGGTTCATACTAAATACTGGATTAGTGCTTTTATCCAAACTATTTGA
ATTAGAAAATTCTTTCTTGATAAAAGTCAAGTTCAGTGGCTCATCTGAGGTTTCAGATGA
TGAAGAAACACTGTTGTGGTCTACGTTGACACTATTGGATTTTGTTTTGTTCTTTGTAGC
TATAATATTTTTGGGTTCCTTCATTTGTTTTGGTAAAGACAAGTCCAAAGGTTCAGCCTG
AAGCTCCTCAGAGGAGAAACTGTTTGGAGTGTAAGAACTACTATGGGAGTTTTTAGAAGA
TGTGGAAGAAAGATTTAATGGTGAAGGAGTATTGCTCCTGGAGTGGTCCAATTTCTCAGC
TGCTTTAATACTGGTAAAGTGAGTAGGTTTTGTTAGCCTGAGAGGAGTATCACAGTTAGT
AACACTATTATGAAGTTCAGCTATAGATGGTGATGTTATAGAGTCCACAGGCTTTATAGG
GGATCTGGCTGACAAAGAGTCTTTCGTGGGAGTATTGTTGGTGGCAGCCAGCACCACCTT
GGCATTGCTCCTTTCCAGGGGTGGCGACCTGGAAGTTGTGTGTTGGTAGACTTTTCGTTG
TTCGAACCATTCCTTCACAAATTCCTGAGGAAGGCCAACAGCAATGGAAATTTTCAGTAG
TTCATCAGAAGTGGGCTCCATATTCATAGCAAAATATGCTTTAAGTACAGACATATGGTC
CTTGTATGGATTGATAGGGCTAGCCATTCCTTTCTCAGAAAGTACAGATGATAGGAGGAG
AGCTTGCTTATCAAAAATCACCCCTGGTTTGCTAGGAACCATGTTTTCGTGAGGTTGGAG
GACTGCCTTGATTTCTTCATTCATCTTACAGAGGTACCGTTCATGTTGATGCAGGGGAAT
GGGTCCAGGAAAACTTTCTTTACAGAATTGGCATGAAAATGGAGTGGGTATATTATGATT
TTCTATCATTTTATCTTCAGTCACCAAATCTATTAGGGTGCGTAGTTTTTCTTTCTTAAA
GTTGTTGAGCTGCCTCCTTGAATCTGTAGTCAAGCTCTGGAGACAAGCTTTGGCTTCATT
GACTTTTTCTAAAGTGTAGTCAATAATACTTTTAGTGGCACCATTATGACTGACTACTGG
AAGACCCACAGGTGGAATACCAGGAGAAGTAATGCCTTGTTCCTCTGGCTGAGAACATGG
GTCCTTCATGTGGTAACCTTTCAGCTTGGAAATTTCTTCAGTGCCACAGTCCATTTTCTG
CCTAGAAACGGTGTTGTCTACAATCTGTAGGACTTTCTGCACTTCACTTAAATTGCTCCC
CATTGCTGGGAATCCAAGTAAAGGTGCTTCCATTCCTACCACTAAGTGCTGCATTGGACT
TTGAGTAGAAGCATGGACTCCTATTGGGCTGGTTGCACCAAGTCTACCGTTCATAAAAGG
ACTAGCACCAGTGAACCCATGGGTTGCCATCAGAACTTTATATTCATTAAAGTCTAGTGG
TTCTGTTTTGATTTTCAGTAAGCCTGACTGCTCAGACATACTAAGTGGTTTTCCATTCTC
CAGCTTGTTTCTCAATTGTGTAATGGCTGAATTAGTAGGTGAGGAAGATACAGAATTAGG
AGAAGAACCCGTCTTGATATTGTTTCTCATTCGACCATTTACAGAGATTAAACCAATACA
TTTCTTGCTGCTGATGTGTGAACTGTATGAACCAGAATGGGAAAAACGTTTCTTGCAATT
GGGACATTCATATGGTTTTTCACCTGTAACGAATTTAAAAAAAGTTAGTAAGAGGCAAAC
CTGTTCTTCAAATATGTAATTTAGCAGCTAATCACATCATGTCTACATATTCTATTTATT
TTATTGAATGATTTCCCCTTCTGCAATTGGAAACTGTCATCCAAAAATTATGCAAATAGC
AGTATGCAAGATTTGTTCCAGCTTTCAT
```
>Node number is a unique identifier for different contigs, length is the number of nucleotides in the contig, and coverage is the average per base nucleotide depth that was used to infer each contig 

The ```Cylindrophis_ruffus_FMNH_258674``` directory contains various log files from the two programs.
 
8. When executed in phyluce, spades compares three kmer values (k = 21, 33, 55) and velvet used (k = 31). If we compare some common statistics of the two assemblies we should see that the results differ. 

The data below were collected using the following phyluce script: 
``` 
for i in spades-assemblies/contigs/*.fasta;
do
    phyluce_assembly_get_fasta_lengths --input $i --csv;
done 
``` 
Spades: 

sample | contigs  | total bp | mean length | 95 CI length | min length | max length | median legnth | contigs >1kb
------------ | -------------  | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | -------------
Cylindrophis_ruffus_FMNH_258674 | 16141 | 7288211 | 451.53404373954527 | 1.308632812051666 | 56 | 5441 | 404.0 | 98

Velvet: 

sample | contigs  | total bp | mean length | 95 CI length | min length | max length | median legnth | contigs >1kb
------------ | -------------  | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | -------------
Cylindrophis_ruffus_FMNH_258674 | 49126 | 12251170 | 249.38260798762366 | 0.4714853349602348 | 100 | 1141 | 237.0 | 1

 
>These differences arise from the different kmer and coverage depth settings which we will explore more in the next module. 
 
</details>

 ## Comparing kmer sizes and coverage depths of *de novo* assemblies using velvet

<details>
 <summary>Click to expand content!</summary>
 
>As we discussed in lecture earlier today, different kmer lengths and coverge thresholds produce different assembly results. We will see examples of this by generating a handful of assemblies in [velvet](https://www.ebi.ac.uk/~zerbino/velvet/). 

1. First let's download velvet 1.2.10: 
 ```
wget https://www.ebi.ac.uk/~zerbino/velvet/velvet_1.2.10.tgz
```

2. Now let's unzip and make the software: 
``` 
tar -xvzf velvet_1.2.10.tgz
```
``` 
rm velvet_1.2.10.tgz
```
```
cd velvet_1.2.10
```
``` 
make 'MAXKMERLENGTH=127' 
``` 
>This should make two executable files ```velveth``` and ```velvetg```

3. Using the *Cylindrophis* *ruffus* sample from [Unit 1](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_01/Bioinformatics_Lab), let's run velvet using the default kmer size:
```
./velveth output_directory/ 31 -fastq.gz -shortPaired Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz -short Cylindrophis_ruffus_FMNH_258674-READ-singleton.fastq.gz
``` 
```
./velvetg output_directory
```  

We should see this output printed: 
 
```
Final graph has 110251 nodes and n50 of 280, max 1111, total 12242834, using 0/221695 reads
``` 
>This is similar to the results we got using velvet in phyluce
 
4. Now let's run velvet using the largest possible kmer size:
```
./velveth output_directory/ 127 -fastq.gz -short Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ-singleton.fastq.gz
``` 
```
./velvetg output_directory
```  
We should see this output printed: 
 
```
Final graph has 7202 nodes and n50 of 311, max 922, total 2403925, using 0/221695 reads
```  
>contigs are on average longer than k = 31, but maximum size is smaller and number of nodes is smaller 
 
5. Now let's run velvet using an intermediate kmer size (k = 80):
```
./velveth output_directory/ 80 -fastq.gz -short Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ-singleton.fastq.gz
``` 
```
./velvetg output_directory
```  
We should see this output printed: 
 
```
Final graph has 33085 nodes and n50 of 283, max 1575, total 8295245, using 0/221695 reads
```  
>This has the longest fragment length of the three kmer analyses... so which one is best? Well... it depends.    
 
 
5. Let's explore the impact of different levels of coverage using (k = 127). 
```
./velveth output_directory/ 127 -fastq.gz -short Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ-singleton.fastq.gz
``` 
```
./velvetg output_directory -cov_cutoff 10
```  
 
We should see this output printed: 
 
``` 
Final graph has 469 nodes and n50 of 438, max 5386, total 199942, using 0/221695 reads
```
>The n50 and maximum length have increased compared to the default settings, this suggests that many of the contigs in the original analysis were short, low coverage contigs that we won't be interested in. However, it does mean we now only have 469 eligible contigs for downstream analysis.  
 
6. Let's try a different coverage depth with the same kmer value: 
 
```
./velvetg output_directory -cov_cutoff 20
```   

We should see this output printed: 
 
``` 
Final graph has 36 nodes and n50 of 423, max 1148, total 15336, using 0/221695 reads
```
>Based on these examples we can see that kmer size and coverage depth are meaningful parameters that impact the composition of contigs from *de novo* assemblies.
 
</details>

## Reference sequence mapping with BWA

<details>
 <summary>Click to expand content!</summary>
 
>Mapping Illumina sequences to a reference can happen in several ways. For example, you could map the *de novo* assembled contigs we discussed earlier in this Unit to a partial genome to improve the genome sequence. Reference-based assemblies can be useful when you have low coverage data (e.g. museum specimen shotgun sequencing) and a good reference genome.  

1. Let's download [BWA](https://github.com/lh3/bwa)
```
git clone https://github.com/lh3/bwa.git
```
2. Now let's make the program:
```
cd bwa
``` 
```
make
``` 
>This should build the executable file ```bwa```
 
3. Let's download a reference sequence to use with BWA. Just like we used the SRA to download FASTQ data in [Unit 1](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_01/Bioinformatics_Lab), we can use the NCBI Entrez Direct UNIX E-utilities to download FASTA files of genome sequences or barcodes for reference mapping.
```  
sh -c "$(wget -q ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh -O -)"
```

4. Now navigate to the edirect directory. Once inside execute the following command: 
```
./esearch -db nucleotide -query "AB179619.1" | ./efetch -format fasta > AB179619.1.fasta
```
>This should download the *Cylindrophis* *ruffus* mitchondrial genome. The resulting FASTA file should look like this:   
 <details>
 <summary>Click to reveal FASTA sequence</summary> 
>AB179619.1 Cylindrophis ruffus mitochondrial DNA, complete genome
GTTATTGTAGCTTACAATAAAGCACAGCACTGAAAATGCCAAGACGGAGCCCTCCCAAATAACATAAAGT
CTTGGTCCTAAACCTAATATTATCTACGACCTGAATTATACATGCAAGCCTCAACATGACGGTGAATAAA
GCCCACACACCAACCGGAGCTGGTATCAGACTCAAAAAAATATCTACGACGCCAAGCAAGCCAATGCCAC
ACCCACACGGGCTCACAGCAGTAATTAACATTGGGCCATAAGTGAAAACTTGACCCAGGAAAGATCCTCA
GGGCCGGTTAATCTCGTGCCAGCGACCGCGGTTACACGATAGGCCCAAGATAATACTAAACGGCGTAAAG
CACGACTAGAAGCAGTTAAACCAATTAAGAATAAAACAAAGCCAAGCCGTAAAACGTCAAAGCCAAACTA
GACACATCTTCTTAATTCACTAACAATTTAACTCGTGAAAACCAGGGCACAAACTAGGATTAGATACCCT
ACTATGCCTGGCCATAACACAGCAACACACAACATAAGTTGCCCGCCAAATAACTACGAGTGAAAACTTA
AAACTTAAAGGACTTGACGGTACTTCACACTCAACCTAGAGGAGCCTGTCTATTAACCGATATTCCACGA
TTAACCCAACCATTTCTAGCCATCCAGCCTATATACCGCCGTCGCCAGCCTACCTTGTAAAAGAAACAAA
GTAAGCTAAACAGTATTACACTAACACGACAGGTCAAGGTGTAACTAATGAAATGGACCAAGATGGGCTA
CATTATCTAACTCAGATAACACGAATAAAGTCATGAAACTAACTCTTAAAGGCGGATTTAGCAGTAAGCC
GAGAATAGAACACTCAACTGAAACCAATGCAATGAAGTGCGTACACACCGCCCGTCATCCCTGTAAACAC
AAATACAACACTCCATAAAACTTAAAAACGCTATAAAACAGGGCAAGTCGTAACATGGTAAGCGTACTGG
AAAGTGTGCTTAGAAACAAAAAGTAGCTTACAACAAAGCATTCGACTTACACTCGAAAGACATTCAAATA
AATCTTTTTGAGCCAAACAAAAATTCAACCACCACACAAAATATACCTAACACATCAAACAAATCATTTG
ACTAAAAAAGTAGATGCGATCGAACACTAAAACAGGCACAAATCAGTACCGCAAGGGAAACAATACACTT
AAGCTAAAAACAGCAAAGATTAACTCTTGTACCTTTTGCATCATGGTTTAGCAAGAACATAAAAGATAAG
AAGAATCAAAACCTAACAAACCCGAAACCAGGTGAGCTACTTCAAAGCAGCCTTTAGGGCACATCCCCCT
CTGTAGCAAAAGAGCGGAAAGACTTAAAAGTAGAGGTGAAATGCCTACCGAACCTGGAGATAGCTGGCTA
CCCAAAAAAGAATATTAGTTCTACTTTAGACCCATAACACAACATCATAGTTCTCTAAAGACATTCAACG
GGGGTACAGCCCCATTGAAACAGGATACAACCTGGACCTGAGAGTAAAAAAACCTAAACCAAACAAGTAG
GCCTTAAAGCAGCCACCTAAAAAAATATCGTTTAAGAATTATAACAATACAAATCCCAAACCAAATCAAA
TACTCCAAGTACACTAAGGGTAAATTTATAACAATAAATAAACTTATGCTAAAACTATTAATAAGATTAA
CCTCTCTACGCACAACTATTCACTAGAAACAGAAACCCCGCTAGCAATCAACAGACCACAAAAGGAATTA
AAACTAACCCACTACACAACCTTTTAACTAACTGTAACCCCGACACAGGTGTGCCTAAAAGAAAGATAAA
ACATTACAAAAGGAACTCGGCAACCAATGACTCCAACTGTTTACCAAAAACATAACCTTTAGCAAAACCA
ATATTAAAGGCAACGCCTGCCCAGTGAAACTAATTCAACGGCCGCGGTACCCTAACCGTGCAAAGGTAGC
GTAATCATTTGTCTATTAATTGTAGACCAGTATGAAAGGCCACATGAGAGTCAGACTGTCTCTTGTAATT
ACTCAATTAAACTGATCTCCTAGTACAAAAGCCAGGATAAACATATAAGACCAGAAGACCCTGTGAAGCT
TAAACTAACCCGTTAAACCCAATAACGGCCACTTTCGGTTGGGGCGACCTTGGAACAAAACAAAACTTCC
AAACAATACTGATACCCTCACATCCAGGCCAACAAGCCACTACACCGACCCAGTCTCACTGATCGATGAA
CCAAGTTACTCCAGGGATAACAGCGCTATCTTCTTCAAGAGCCCATATCAAAAAGAAGGTTTACGACCTC
GATGTTGGATCAGGACACCCAAATGGTGCAGCCGCTATTAAAGGTTCGTTTGTTCAACGATTAACAGTCC
TACGGCGATCTGAGTTCAGACCGGGAGCAATCCAGGTCAGTTTCTATCTATAAAATGCTGTCTTTAGTAC
GAAAGGATCAAGACAACAAAGCCCATACCAAAAGCACGCTTTAACAAACAATATTAACAAACTCAATATT
TCAACACTACTAACATCAAAGCCAAAAACAGGCTAATTAAGGACTAAACCTACTAACAGGCCTATTCCCC
CCTTAATAATATCAAATATCACATCACTTATTATCAATCCTCTATTATATATCATCCCGATCCTAATCGC
AGTAGCATTTCTCACCTTACTAGAACGAAAACTTCTAGGATACATACAACTACGAAAAGGTCCAAACCTT
GTAGGTCCACAAGGACTTCTACAACCAATCGCAGACGGAATTAAACTAGTAATTAAAGAGCCAACAAAAC
CCACACTATCTTCACCAACCCTATTCATACTATCACCAATAATAGCACTCACACTAGCACTAGCAACTTG
AGCACCAATACCCATACCATTCCCACTTACAAACATAAACCTAGGTCTATTATTTATTATAGCTATATCA
GGGATATTCACCTATGCAATCCTATGAGCAGGGTGATCATCGAACTCAAAATATCCACTCATAGGCGCAA
TACGAGCCGTAGCACAAATTATCTCATACGAAGTCACACTGGGATTAATCATCATATCAATAGCTACAAT
CACAGGCGGATATTCACTACAAACATTCACAACCACTCAAGAACCATCTTGACTCCTACTACCATCATGA
CCACTGGCTATAATATGATTCACCTCAACTTTAGCAGAAACAAATCGATCACCATTTGATCTAACAGAGG
GGGAATCAGAACTAGTATCCGGCTTCAACGTGGAATTCTCAGCAGGACCATTCGCCCTACTATTCCTAGC
AGAATACACCAACATTCTAATAATAAACACACTATCAGCCATAATATTCTTTAATCCAGGAACAATAAAA
ACCCCCACACTATTTACAATTAACTTAATAACAAAAACAACCATCCTCACCATTATATTTCTCTGAATCC
GCGCATCATACCCACGATTCCGATATGACCAACTCATACACCTATTATGAAAACAATATCTACCACTCAC
CCTAGCCATATGTATACTTAACATCTCCACGACCTTTTCATTCCACGGTTCACCACCACAATGGAAGTGT
GCCCGAGACTAGGGACTACCTTGATAGAGTAGATACAGGACCTTCAACCTCACTTCCTAGAAACCGCAAC
CCATTATCACGGTGGGTTGCGGTTTCTACCCACATAAAACAAATACCCCCTTGGACCCCCCCCTACCCCC
CCCATAATATTTATACTAAATTTCACTTCACACACATAATGTAACCTTATACATTAGTCTATATTTCATT
CTATGTATAATCTTACATTAATGGTTTGCCCCATGAATATTAAACGAGAATTTCCCTTTAATTATTTTAG
TCTAGAAATGCTTCTAAACAATTTATTCAGTCACCCATACCGGTCGTTCCATGAAGCACGGAATATACCC
ATTCTTAATAACCATGACTATTCTTTCTCTAGTGGTGTCTCTTAATTTAGTCCTTCCCGTGAAACCCTCT
ATCCTTTCATACAGGCTAACCATTCGACTTCTCACGTCCATAATAATGCCAATCCTCCCTACTAGCCTTT
TCCAAGACCGCTGGTTACACTCTCAAGTTCATCTCAATGGCCCGGAACCATCCCTCCCTACTAGCTTTTT
CCAAGGCCTTTGGTCGCACCCTTTATTGTGGTTCATTAATCTCATGATCTGATCACATATGCCAGTCCAC
CTTTGGTTGGTCTTTTTTTCCTGTACCTTTCACCTGACTACCATATATGCAAAACACGTAGGTAGGTTAT
TTATCCAGGTGGAGCACCATGGTTGGTTTTACACCCTCCCTTATGGATACATTCTATTAATGTTTGATAG
ACATACTCTGCAACCCAAAAATTTTTTTTCCACCACTTAATTCAACCACACAATTTCCACCCCAAATAAA
ACAAACTAACTTTCCATAACAACCAATCTAAAAAACCATCAACTCATGCTCACCTCACTCTTTTTTTATA
TAAACTTTTACCTCAAAACTAAAATGAACGGACCTTTTATATTACCAGCCCAAATAGCAACCCCCTTCAC
ATAAATCGCTTCAAACTAAAAAGTTGACTTCACATATAACCAACCAAAAAATCAACTCATACTCACCTCA
CTCCATTTTTATACAAACTTTAACTTTAAAACTAAAATGAACTAACCTTTTATATCAATCAACCCAGAAA
GAGGAGGCCTACAATCCACGCCCTCCCATATAAACTTTAAACCAAACAATTTATATAGACCAATCAATCC
CATTAAGGTAGCAAAACCAGGCCATGCAATAGGCTTAAAACCTCAACACAGATGTTCAAATCATCTCCTT
AATACTAGAAGGCCAAGAATCGAACTTGAACTAAAAAGCCCAAAACTTTTAGTACTACCATATATACTAC
CTTCTAAACAGTAAAGTCAGCTAAACAAGCTATCGGGCCCATACCCCGAAAATGCCCCACGGCCTCTACT
AATCAACCTAACATCTTGAATCACAATCTCAACCAGCATTACACTAAGCACAATCCTAGTCGCTACATCA
ACACACTGAATCATAGCATGAACCTGCCTAGAAATCAACACTATATCCATAATCCCAATCATCTCAAAAC
CACATCACCCCCGGGCCACAGAAGCCGCAACAAAATACTTCTTAACCCAAACTATAGCCTCGATAACCAT
ACTATTTGCCACAACAATAAACGCTCTTAACACCTCAAACTGAGAAATTACACTCATAACCGATAACACA
ATAATCGCCCTAATAACACTAGCCCTTATAATAAAAACTGCAGCAGCTCCATTCCACTTCTGACTACCAG
AAGTATCACAAGGAGCATCAACCTTAACAAGCCTAACAATCCTAACATGACAAAAACTAGCACCATTATC
CATCCTACTAACACTCTCTAACAAAATTAACCAAACCATTCTCCTATCCATAGCAATCCTATCAATTATA
ATCGGTGGCCTAGGAGGCCTAAACCAAACACAACTACGAAAAATCATAGCTTTCTCATCAATCGCCCACA
CAGGGTGAATCCTATCAACAATAGCAATCGCACCAAACATCTCAACACTGACCCTAACAGTGTACATTAT
AACAACTACCCCCATCTTCATCCTACTAAACTCAACTTCAACCACAACAATCAAAGACATAGGAACAATA
TGAACAATGTCCCCACAACTAGCATCAGCCATCACACTAACAATCTTATCACTAAGCGGCCTACCTCCAC
TAACAGGATTCATACCACAATGACTTATCCTAAACAAAATAGTCATATTCAACATGACTACTGAAGCCGC
CATTATAGCCATAATATCCCTACTAAGCCTATATATATACCTACGACTTACATACATCTTTTTAATAACC
CTTTCACCCCACACCAACATTATGACAATAAAATGACGAACAAATCACAAAACACTGACTCCGATCACAT
CAACACTAATAATCCTATCCACCATAATACTACCAATAACACCCATACTTAGAGAAACTTAAGTTATACA
AACTAGAGACCTTCAAAGTCCCCAAAAAAGACATACTTTAGTTTCTGAGAGCTTGCGGCTACCCACATCT
TCTGCTTGCAGAACAGACATTTTAATTAAACTAAAACTCTCCTAGACTAGTGGGCCTCGATCCCACAAAA
AACTAGTTAACAGCTAACCGTCCAAACCAGCGGACTTTAGTCTACTTCTCCGTTTTACGGGAAAAAACGG
AGAAGCCCCGGGCAGAACGCCTTCTTCAGATTTGCAGTCTGACATGTAACACCTCGGGACCTGGCAGCAA
AGGTTTAATCCTTATAAGTAAATTTACAGTTTACCGCTATATCAGCCATACTACCTGTGTTCATCACCCG
TTGACTATTCTCAACAAACCACAAAGACATCGGCACCCTATACCTCCTATTCGGAGCATGGTCTGGTCTA
ATCGGAGCCAGTTTAAGCATTCTAATACGAATAGAGCTCACGCAGCCCGGATCACTATTCGGCAGCGACC
AAATCTTTAACGTACTAGTCACAGCTCACGCATTCGTAATAATCTTTTTCATAGTAATACCAATTATAAT
TGGAGGGTTCGGAAACTGATTGATTCCTCTAATAATTGGGGCCCCAGATATAGCTTTTCCACGGATAAAT
AACATAAGCTTCTGACTTCTGCCGCCAGCACTTCTATTACTTTTATCCTCTTCATATATTGAAGCCGGTG
CCGGCACCGGCTGAACTGTTTACCCACCCCTCTCAGGAAACATAGTACATTCTGGCCCCTCAGTAGACTT
AGTAATTTTTTCCCTTCATCTAGCAGGAGCATCATCCATTCTCGGAGCCATCAACTTTATTACCACATGC
ATTAACATAAAACCAGCATCTATACCAATATTTAACATCCCCCTATTCGTATGATCTGTTATAATCACCG
CAATTATACTCCTACTCGCCCTACCAGTTCTAGCGGCAGCAATCACAATACTACTAACAGATCGAAACCT
CAATACATCCTTTTTCGACCCTTGCGGAGGAGGCGATCCAGTCCTATTCCAACACCTATTCTGATTTTTT
GGACATCCGGAAGTCTATATTCTAATTCTACCCGGCTTCGGCATCGTATCAAGCATTATCACATACTACA
CTGGAAAAAAGAACACCTTTGGATATACAAGCATAATCTGAGCAATAATATCAATCGCCATCTTAGGATT
CGTAGTATGAGCACATCATATATTTACAGTTGGACTAGATATCGATAGTCGAGCCTATTTTACAGCAGCT
ACTATAATCATCGCAGTACCAACTGGAATCAAAGTGTTCGGCTGACTAGCCACACTAACTGGCGGACAGA
TCAAATGACAAACCCCAATCTATTGAGCTCTTGGATTTATCTTCCTATTCACCGTCGGGGGGATAACAGG
AATCATTCTAGCAAACTCATCACTAGATATCGTACTACACGATACATACTACGTAGTAGCACACTTCCAC
TACGTGTTGTCAATAGGAGCAGTATTCGCTATTATGGGTGGACTAACTCACTGATTTCCATTATTCACAG
GTTATTCACTGAATCAGACCTTAACAAAAACTCAATTTTGAGTAATATTCGTAGGGGTTAATATAACATT
CTTCCCACAACATTTCCTAGGACTTTCCGGAATACCACGACGATACTCAGACTTCCCAGATGCCTTCACC
CTATGAAACACAGTATCATCAATCGGATCCACTATTTCACTAATCGCAGTATTTATATCTCTATACATTG
TATGAGAAGCTATAACATGCAAACGAAACCAGCTTATCCCACTAGGGAAAAAAACCCATGTAGAGTGATT
CTACGGCACCCCGCCCCCATATCACACTCACACAGAACCAACCTTTATACTAAATAACACAGCCCACCAC
CAACACTAATAAACACACAGAAGAGGCAGATTTAAACCACCATCCGCTGATTTCAAGTCAACTGCATCTA
TTAATGCTTTCTCCTCGAGAACCTAGTAAATATATTACATGGCTTTGTCATAGCCAAATAACAGCACCCT
GTGGTTCTCAGTGCCATATGCAACCCAACTCTCACTACAAGAGGCCACAGGACCAACAATAGAAGAAGTT
GTATTTCTACACGACCACGTACTAATACTTACTTGTCTAATAACTCTAGTCATTCTCCTATTCGCTATAA
CAGCCACAATAGCTACCGTAACTCACAACGACCCAACAGAAGAAGTAGAACAGCTGGAAGCAGCCTGAAC
GGCTGCTCCCATTATAATTCTCATTCTAACAGCCCTCCCATCAGTTCGATCATTATACCTTATAGAAGAA
GTATTTGATCCATACGTAACTATTAAAGCAACTGGCCATCAATGATACTGAAACTACGAGTATACAGACG
GAGTCAACATTTCATTCGACTCTTATATAATCCAAACCGAAGATCTACCAAACGGCTCACCCCGATTATT
AGAAGTAGACCACCGTATAATCATACCAGCCAACCTTCAGACCCGAATTGTAGTGACTGCAGAAGATGTT
CTACACTCGTGAGCAGTCCCCTCATTAGGAGTAAAAATCGACGCAGTACCAGGACGACTAAATCAACTCC
CACTTTCCACTTCACGAACAGGTGTATTCTACGGCCAATGTTCAGAAATCTGCGGGGCAAATCATAGTTT
CATACCAATCGTAGCAGAAGCGGTTCCACTAAAATATTTCGAACATTGACTTTCTTCAGAAAATTCATCA
AGAAGCTTTTACAGCATTAACCTTTTAAGTTAAAGATAGAGATTGACCTCCTTGGTGACATGCCCCAACT
AGATGTCGTATACGCCCTTCTAACCTTTATCTGGGCCTGGTCCACTCTTGTCGTTATAATAATTAAGATC
CGAACTATCATACTAAATAAAGAATTAGAAAAAAAACCAATAACAAATGATCAACCAGGACTTATCTGAA
ACCTACCATGAACCTAAATATATTTGAACAATTTGCAAGCCCAGAAATCCTACTACTCCCCACCAACATC
ATATCAATACTAATCCCGATTCTTCTAATCTATACTAAGCCAACCCTACTAGGAAATCGAATCTCAACAA
TCCTAAACTGATTCCTAAAAACAGCCCTATCAAACATAATCAACCAGCTTTCTTTAGCTGGCCAAAAATG
ATCACGATTTTTAATAAGCCTCCTACTACTTATTCTATTATCTAACCTACTAGGATTATTACCCTATACC
TTTACCACCACCTCGCAACTATCAATAAATATAGCATTCGCTATTCCAATATGACTGGGAACAGTAATCA
CAGGTTTAACAATAAAAACATCATCCTCCCTAGCACACATACTACCAGAAGGGTCACCAACCCCACTCAT
TCCATTCATAATCCTAATTGAAACTATTAGTCTACTTATACGCCCAATCGCACTAGGAGTGCGACTGACA
CCCAATATCACAGCAGGCCACCTTCTCATAACTATAATCAGCTCTGCTGCCCTAAGCTTCATTAACATAT
ACAACACCCTATCTATTATAACAATAATATTATTAATACTACTCACCTTACTAGAAATGGCCGTAGCTTG
CATCCAGGCCTACGTCTTTATCCTCCTAATCACCCTTTACCTTCAAGAAAACACATAATGACCCACCAAC
AGCTTCACCAATACCATATAGTCGATCCAAGCCCATGGCCCCTAACAGGGGCCATGGGCTCGTTGCTACT
AGCTTCAGGACTAGCCCTATGATTTCACACCAACACCACAACAGTACTAAAACTTGGCATACTAACTATA
ATTCTAACCACCATCCAATGATGACGAGATGTTATCCGGGAGGGAACCTATCAAGGACATCACACAAAAG
GTGTCCAAAAAAATCTACGATACGGAATAATACTATTCATTGTATCAGAAGTACTATTTTTCCTAGGGTT
CTTCTGAACCCTATACCACGTCAGCCTATTCCCAACTCCGGAACTGGGCGCAGAATGACCTCCAACAGGC
ATCACCCCACTAAACCCTCTCGAAGTACCTCTACTTAACACTGCCGTACTTCTTTCATCCGGAGCAACCA
TCACATGATCGCATCATACAATAATAAAAGGAAATAAAACAGAATCAACATACGCCCTAATAATCACTAT
CGCACTGGGAATTTATTTTACAGCTCTACAACTATCAGAATATATAGAAACCCCATTCACAATTTCAGAT
AGTGTTTATGGTTCAATCTTCTTTGTGGCAACAGGATTCCACGGCCTACACGTCATAATCGGAACAACAT
TCCTAATTGTTTGTTTGACCCGCCTTATCCAATACCACTTTACAACAACACACCATTTTGGATACGAGGC
AGCTATCTGATACTGACACTTCGTTGACGTAGTGTGACTATTCCTATTCGTCTCAGTTTATTGATGAGGC
TCATATTTCTTTAGTATAGTAGTACAAATGCCTTCCAAGCATTTCGCCCCGCCAGGGAAGAAATAATAAA
CCTGATCGCCCTACTAATAATTTCACTATCCGTAGCAACACTCTTATATATAATTAACACCTACATAGTC
ACCAAACCAGACATCAACAAACTATCACCATACGAATGCGGATTCGATCCACTAGGAGACGCACGATCAC
CAATCTCAATCCAATTCTTCCTAGTAGCCATTCTATTTATCCTATTTGACCTAGAAATTATCCTACTACT
CCCCATCCCATGGAGTATAAATACTAACCCCCCAACTACCACAATCACCTTGACAATAATACTACTTATC
TTACTAACACTAGGGTTAGCATACGAATGATATCAAGGCGGACTAGAATGAGCTGACTCCGGAAGTAGTC
TACACCAGATATCTGATTTCGACCCAGAAGACCTTAATAATTAAGCTTCCGTAATGGAACTCACTAAGAT
CACATTATCAATAATTTTCCTTATAGTCATACTAGGACTATCCACACAACACAAACACCTCATACTAGCC
CTAATGTGTATCGAAGCAATAATACTAACCTTATTTATAATACTAGTACTATATACTACAACCTCACTAA
CCATATCCCAAACCCCCATACCAATAATCCTTTTAACTATCTCCGTATGCGGAGCCTCTGTAGGACTTAG
CCTAGTCGTAGCAACCACACGAACTCACGGTAACGATTTCCTAAAAAACCTAAACCTCCTATAATGCTAA
AAATTATTATCATAACAACAATAATAATTCCAACAACCCTACTACTAAAACCAAAGATTCTTTACCAAAC
GACTATCTCATACTCATTCATTCTAACCCTTATAAGCCTGATACTCCTAAAACAAAACACGCACTTAAAA
CCCCTATCAAACTTATACCTAAGTCTAGACTCAGTTTCAGCCCCACTACTATCACTCTCATATTGACTTC
TACCCCTAATAGTTATTGCAAGCCAACATATAATAAACTCAGAACCAATACAACGCCAACGAACATTCCT
AATAACTATTATACTCTTACAAACCCTAATTTCACTAACATTCACAGCATCTAATTTAACGTTAATATAT
ATTATATTTGAAGCCACCCTAATTCCAACACTAATTATCATCACACGATGAGGCCAACAATCCGAACGAT
TAACAGCAGGAACCTACTTCATTATATACACCCTAACAACATCAATACCCCTACTAATAGCAATCTTATT
CCTAAATAACATATCAAATACCCCAACCCCATTTTTCATAACAACCGAAACAACAAACACATGAACCAAC
CTTATCCTATGATTAGCATGCCTAACTGCCTTCCTAGCAAAAATACCCATATACGGGCTACACTTATGAC
TACCAAAAGCCCATGTAGAAGCGCCCATCGCAGGCTCCATAGTCCTAGCAGCCATTCTACTAAAATTGGG
CGGATACGGCATAATCCGAATAATACAGATCCTCCCGGTCATAAAAACAGACGCATTTATACCATTCATT
ATCCTCTCACTATGAGGAGCAATCCTGGCCAACCTAACCTGTCTACAACAAACAGACCTAAAATCCCTAA
TCGCATACTCATCTATCAGCCACATAGGCCTAGTAATCGCGGCTATCATAATCCAAACCCCATGAAGTCT
AGCAGGGGCAATAGCACTAATAATCGCCCACGGATTCACCTCATCAATACTATTCTGTCTAGCCAACATC
ACCTATGAACGTACACATACACGAATTATAATACTAACTCGAGGTTTTCACAATATTCTGCCCATGATGA
CAATCTGATGACTTTCCGCTAACCTAATAAACATTGCCATCCCACCAAGCCTGAACTTTACTGGAGAACT
TCTAATCTTATCATCCCTATTTAATTGATGCCCAACAACCATCATCATCCTGGGCCTATCCATACTAATT
ACTGCAACATACTCCCTACACATATTCCTAACTACACAAACAAGCAAATCTTCACTAAATCTACCAACTC
AACCAACCCACACTCGAGAACATCTCCTCCTATTCCTTCACATCACCCCGCTCATATTAGTATCAATTAA
ACCAGAACTTATCCTAAGGGTGTCTGTAATTTAAAAAAAATATCAAGCTGTGACCTTGAATATAGAGACC
TGCTCTCATACACCGAGAATGTTATAAGACCTGCTAACTCTTCTATCTGGACCTAAACGGCCTGCCTTCT
CTTCTATTAAAGGATAATAGCCTTCCATTGGTCTTAGGCACCAAAACTCTTGGTGCAAATCCAAGTAGTA
GAATATGAACACCATCACTCCCACACTAACAGTAACCATTTTTATATCTCTAACGCTCCTAATCATAAAA
CTACTCATTAAAAAAACTCAAAACTTAACCCATACAAAACACATTATAATACTTATATTCATCATTAGTC
TAATTCCACTTAGCCTCCTACTGAATAACGAAAACGAAACCATACTATCATCCCCCCCAATTATTTATAT
AACAACAACAAACATCAGCATCAGCCTCATCTTAGATACTTTCTCATTAACCTTTATCCCAGTCTCTCTA
TTTATTACATGATCTATCGTAGAGTTTTCAATCTGATATATGTCAACCGACCCATACATTAATAAATTTA
TCAAACATCTACTCACCTTCCTAATCGCTATACTCATTATTATCACAGCGAACAACATAATCCAGCTCTT
CGTGGGGTGAGAAGGTGTAGGAATCATGTCCTTTCTACTAATCGGATGATGATACGCCCGATCAGACGCC
AACAGCGCAGCCCTACAAGCCATTATCTACAATCGAATTGGAGACATCGGCCTCATCATAACCACTGCAT
GACTAATATCATCCTCATCAATAAACATACAAGAACTATTCACACAACACGAAACAATAAGCATTATCCC
ATTAATAGGCTTAGTAGGCGCAGCCGCAGGAAAATCAGCCCAATTCGGTCTTCACCCATGACTACCTGCA
GCCATAGAAGGCCCTACACCAGTATCAGCCCTACTCCACTCAAGCACAATAGTTGTAGCAGGCGTATTTC
TACTAATTCGCCTACATCCAATCATACAAAACAATATCACATCAACAACCTGCCTAATCCTGGGGGCCAC
AACAACTCTATTTGCAGCCGCCTCAGCAATTACCCAACACGATATTAAGAAAATCATCGCATTATCAACA
ACAAGTCAACTAGGACTAATAATAACAATAATCGGACTAAACCAACCAAAACTAGCCTTTCTTCATATAG
CTATACACTCCTTCTTCAAAGCACTATTATTCCTATGCTCAGGATCATTCATCCACAACCTTGGTGGAGA
ACAAGACATCCGAAAAATAGGAAACCTAAACAAAAACCTACCCATAACATCATCAATAATCACAATCGCC
AGCCTATCACTCATAGGCATACCATTCTTATCCGGATTCTACTCTAAAGATACGATCATCGAAACAATAA
CTATCTCCCACATCAACTCATGAGCTCTTATACTTACACTGGTCGCAACAATACTATCTGCAATATATAG
CATACGAATCATTAACTTTACATTAACCAACTTCCCACGCACCAAACAGAAAATTCACCAAGAAAATAAA
ACACTAGCCAAACCAACCCTACGTTTAACCCTGGGCTCAATCTTAGCAGGTACAATAACAAAACTATCAA
CCTTACAAACAACATCAACTATAACTATACCAACAACAATTAAACTAAGCGCACTAACAATCACCATCGC
AGGCATTCTACTTTCAACAGACCTAATATTTCTATCACCCCACCAACCCCCTAAAAAACCAAAAACGCTA
ACCCTGTTTTTCAACCAACTGGCCTTCTTCAACATAATCCATCGAGCAGCACCAATAAAAACATTAAAAT
TCGGACAACAAATCTCAACAGAACTAGTAGACCTATGAGCCCTAGAAAACTACGGACCAAAAGGCCTATC
CAACTCAACTATTAAAATAATCCACACAACCACACAACAAAAAAATTTAATCAAAAACTACCTAACCACG
TTCACACTTACTATTATCCTCACATTAGTCCTTATAATATCTAAAAGACCGCAACCCCCCCAAACGATGT
CAACTTAAAATTACTAAAATAGAAAATAACGTCACAAGCAAACCCCACACACAAACCATCAACCCAGCCC
CGCCACTAAAATAAAGAACTCCAAAACCATTAACCTCAACACAAGCAAAACTATCCTGATCCATCACAAC
TAATCACCCATCTAACCCAAAAACCACCGCAATCCATAAATAAAAAAAGGCTGCAACAAATACAGACACA
CATACAAATATCTTAGAAACAAACAACTTAAAAACATTATCTACACCCTTCTCAACACTAATACAGTAAC
TAAATACAACTACCAAACCCCCCAAATACACAATATAAGTCACTAGCGCAACAAAAGTACGACTAATAAA
ACACATAAAAACACAACACAAAAACGCAAACCCCATTAAAGCAATAACCCCATAATAAGACACCGGCGTC
GTCCCCAATACTACCGCTACCACAACCATCAATATTATAATCAAACAAAAAACATAATCTATAATAAACA
TAATTTCTGCTCAACTTCTCGAGTCCTGCGGTCTGAAAAACCACCGTTGTATATCAACTACAAAAACATG
CCCCACCACTATATCCTAACATTATTTGGCTTACTCCCAGTAGCAACAAACATTTCAACATGATGAAACT
TCGGCTCAATACTACTATCATGTCTCGCTATTCAAATTATAACTGGGTTCTTCCTAGCCGTCCACTACAC
AGCAAACATTAACTTAGCATTTTCATCAATCATCCACATCTCCCGAGATGTCCCACATGGCTGAATAATA
CAAAACCTTCACGCCATCGGGGCCTCCATATTCTTCATCTGCATCTACATCCACATCGCACGAGGCCTAT
ACTACGGCTCCTACCTTAACAAAGAGACCTGAATATCAGGCATCACCCTACTTATTACATTAATAGCTAC
AGCCTTCTTCGGATATGTACTCCCGTGAGGACAAATATCATTCTGAGCTGCAACTGTTATCACAAACCTA
CTCACCGCCATCCCATACCTAGGAACATCCTTAACAACCTGATTATGAGGCGGATTCGCAATCAATGACC
CAACCCTAACCCGATTCTTCGCACTCCATTTTATTCTACCATTTATCATTTTAGCCCTATCCCTAATCCA
TGTAATCCTCCTACACGAAGAAGGCTCAAGCAACCCCCTGGGCACCAACCCAGACATCGACAAAATCCCA
TTCCATCCTTATCACTCCTACAAAGATTTTCTCCTACTAACAATAATAATCATTATTCTCCTTATTACAA
CTTCATTCCTACCAGATATTTTTAACGACCCAGACAACTTCTCAAAAGCCAATCCCCTAGTAACCCCACA
ACACATTAAACCAGAATGATACTTCCTATTTGCCTATGGCATCCTTCGATCAATCCCAAACAAACTTGGG
GGGGCCCTAGCCCTAATCATATCAATCATAATTCTATTTACCCTACCATTTACTCACACAGCCCGTCTAC
GACCAATAACATTTCGGCCTATAACACAAGTAATATTCTGAACACTAGTATCCACATTTACTGTAATCAC
ATGAGCAGCCACAAAACCTGTAGAACCACCATTCACAATAATCAGTCAAATAACCTCAACAGCTTACTTT
ACATTCTTCATTATAAACCCATTAATAGGCTGAACAGAAAACAAAATAACAAAAACTATCGAATGCTCTA
ATAGCTTAAAACCATAAAGCATTGTTTTTGTAAACCAAAGCTGGGTAACCCCCTTAGAGCATCAAAGAGA
AAATACTTCATCCCTGGTCCCCAAAACCAGAATTTTTAACTTAAACTACTCTTTGGAAACCGCAACCCAT
TATCACGGTGGGTTGCGGTTTCTACCCACATAAAACAAATACCCCCTTGGACCCCCCCCTACCCCCCCCA
TAATATTTATACTAAATTTCACTTCACACACATAATGTAACCTTATACATTAGTCTATATTTCATTCTAT
GTATAATCTTACATTAATGGTTTGCCCCATGAATATTAAACGAGAATTTCCCTTTAATTATTTTAGTCTA
GAAATGCTTCTAAACAATTTATTCAGTCACCCATACCGGTCGTTCCATGAAGCACGGAATATACCCATTC
TTAATAACCATGACTATTCTTTCTCTAGTGGTGTCTCTTAATTTAGTCCTTCCCGTGAAACCCTCTATCC
TTTCATACAGGCTAACCATTCGACTTCTCACGTCCATAATAATGCCAATCCTCCCTACTAGCCTTTTCCA
AGACCGCTGGTTACACTCTCAAGTTCATCTCAATGGCCCGGAACCATCCCTCCCTACTAGCTTTTTCCAA
GGCCTTTGGTCGCACCCTTTATTGTGGTTCATTAATCTCATGATCTGATCACATATGCCAGTCCACCTTT
GGTTGGTCTTTTTTTCCTGTACCTTTCACCTGACTACCATATATGCAAAACACGTAGGTAGGTTATTTAT
CCAGGTGGAGCACCATGGTTGGTTTTACACCCTCCCTTATGGATACATTCTATTAATGTTTGATAGACAT
ACTCTGCAACCCAAAAATTTTTTTTTCCACCACTTAATTCAACCACACAATTTCCACCCCAAATAAAACA
AACTAACTTTCCATAACAACCAATCTAAAAAACCATCAACTCATGCTCACCTCACTCTTTTTTTATATAA
ACTTTTACCTCAAAACTAAAATGAACGGACCTTTTATATTACCAGCCCAAATAGCAACCCCCTTCACATA
AATCGCTTCAAACTAAAAAGTTGACTTCACATATAACCAACCAAAAAATCAACTCATACTCACCTCACTC
CATTTTTATACAAACTTTAACTTTAAAACTAAAATGAACTAACCTTTTATATCAATCAACCCAGAAAGAG
GAGGCCTACAATCCACGCCCTCCCATATAAACTTTAAACCAAACAATTTATATAGACCAATCAATCCCT 
 </details>
 
5. Now let's copy this to the ```bwa``` folder along with the cleaned fastq reads from *Cylindrophis* *ruffus* from [Unit 1](https://github.com/nhm-herpetology/museum-NGS-training/tree/main/Unit_01/Bioinformatics_Lab):
 
``` 
cp AB179619.1.fasta /home/jefs/NGS_course/Unit_2/bwa
```
>Now naviagate to the phyluce directories from Unit_1
```  
cp Cylindrophis_ruffus_FMNH_258674.contigs.fasta /home/jefs/NGS_course/Unit_2/bwa
cp Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz /home/jefs/NGS_course/Unit_2/bwa
cp Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz /home/jefs/NGS_course/Unit_2/bwa
```

6.Now let's run BWA using the mtDNA genome of *C. ruffus* as the reference sequence. First, we need to index the reference sequence. 
``` 
./bwa index AB179619.1.fasta
```
 
7. Next, we will map the contigs from spades to the reference sequence: 
``` 
./bwa mem AB179619.1.fasta Cylindrophis_ruffus_FMNH_258674.contigs.fasta -t 4 > bwa_mem_alignments1.sam
```
 
8. Let's also align the cleaned reads to the mitochondrial genome: 
``` 
./bwa mem AB179619.1.fasta Cylindrophis_ruffus_FMNH_258674-READ1.fastq.gz Cylindrophis_ruffus_FMNH_258674-READ2.fastq.gz -t 4 > bwa_mem_alignments2.sam
``` 
>At this point we now have reference mapped the contigs from the spade assembly and the cleaned, raw reads to the mtDNA of *Cylindrophis* *ruffus*. Remeber these sequences are from a targeted sequence capture (TSC) experiment for ultraconserved elements used in Streicher & Wiens (2016), so we don't expect them to have very many mtDNA reads (as these will have mostly been washed away). Nonetheless, we will see in the next module that there is evidence of mtDNA 'bycatch' in this TSC sequencing library.
 
</details>

## Using SAMtools to view reference maps

<details>
 <summary>Click to expand content!</summary>

>Now we will use [SAMtools](https://github.com/samtools/) to view the mapped sequences. 
 
1. Let's start by downloading and making the program: 

``` 
wget https://github.com/samtools/samtools/releases/download/1.3.1/samtools-1.3.1.tar.bz2 -O samtools.tar.bz2
``` 
``` 
tar -xjvf samtools.tar.bz2
``` 
``` 
rm samtools.tar.bz2
``` 
``` 
cd samtools-1.3.1
``` 
``` 
make
```
>This should give you an executable ```samtools``` file
 
2. Next, let's copy over the reference FASTA and SAM files from the last module:
 ```
 cp AB179619.1.fasta /home/jefs/NGS_course/Unit_2/samtools-1.3.1
 cp bwa_mem_alignments1.sam /home/jefs/NGS_course/Unit_2/samtools-1.3.1
 cp bwa_mem_alignments2.sam /home/jefs/NGS_course/Unit_2/samtools-1.3.1
```
3. Now let's convert the .sam files to .bam files 
``` 
./samtools view -S -b bwa_mem_alignments1.sam > UCEs1.bam
./samtools view -S -b bwa_mem_alignments2.sam > UCEs2.bam
```
```
./samtools sort UCEs1.bam -o UCEs1.sorted.bam
./samtools sort UCEs2.bam -o UCEs2.sorted.bam
``` 
``` 
./samtools index UCEs1.sorted.bam
./samtools index UCEs2.sorted.bam
``` 
```  
./samtools tview UCEs1.sorted.bam AB179619.1.fasta
./samtools tview UCEs2.sorted.bam AB179619.1.fasta
``` 
 >Press ESC to exit the tview screen
 
 </details>
 
**Helpful Links** 
> [velvet](https://www.ebi.ac.uk/~zerbino/velvet/) | [Spades](https://cab.spbu.ru/software/spades/) | [BWA](https://github.com/lh3/bwa) | [WinSCP](https://winscp.net/eng/download.php) | [PuTTY](https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html) 

